src/test/java/com/jcabi/log/UnitTestAppender.java

```diff
@@ -0,0 +1,82 @@
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met: 1) Redistributions of source code must retain the above
+ * copyright notice, this list of conditions and the following
+ * disclaimer. 2) Redistributions in binary form must reproduce the above
+ * copyright notice, this list of conditions and the following
+ * disclaimer in the documentation and/or other materials provided
+ * with the distribution. 3) Neither the name of the jcabi.com nor
+ * the names of its contributors may be used to endorse or promote
+ * products derived from this software without specific prior written
+ * permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT
+ * NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND
+ * FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL
+ * THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT,
+ * INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+ * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+ * STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED
+ * OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package com.jcabi.log;
+
+import java.io.ByteArrayOutputStream;
+import java.io.UnsupportedEncodingException;
+import org.apache.log4j.PatternLayout;
+import org.apache.log4j.WriterAppender;
+
+/**
+ * Log4j appender for unit tests. Normally, we could use
+ * <a href="http://projects.lidalia.org.uk/slf4j-test/">slf4j-test</a>, but we
+ * have log4j in the classpath anyway, for {@link MulticolorLayout}.
+ * @author Mihai Andronache (amihaiemil@gmail.com)
+ * @version $Id$
+ * @since 0.18
+ */
+public final class UnitTestAppender extends WriterAppender {
+
+    /**
+     * OutputStream where this Appender writes.
+     */
+    private final transient ByteArrayOutputStream logs =
+        new ByteArrayOutputStream();
+
+    /**
+     * Ctor.
+     * @param name The appender's name
+     */
+    public UnitTestAppender(final String name) {
+        super();
+        this.setName(name);
+    }
+
+    /**
+     * Prepares the appender for use.
+     */
+    public void activateOptions() {
+        setWriter(createWriter(this.logs));
+        setLayout(new PatternLayout("%d %c{1} - %m%n"));
+        super.activateOptions();
+    }
+
+    /**
+     * Return the logged messages.
+     * @return String logs
+     * @throws UnsupportedEncodingException If the bytes cannot be converted
+     *  to String using UTF-8
+     */
+    public String output() throws UnsupportedEncodingException {
+        return new String(this.logs.toByteArray(), "UTF-8");
```

Comment:
@amihaiemil actually it doesn't matter here, but it's possible to use https://docs.oracle.com/javase/7/docs/api/java/nio/charset/StandardCharsets.html#UTF_8 instead of string here